Code review

IraCherginyshka opened the pull request.

IraCherginyshka committed:
Create an application with Angular CLI (#2)

* Create an application with Angular CLI

* Change 'proj' => 'AwesomeTetrisGame' in angular.json, karma.config.js and package.json files

* Resolve: rename project name in angular.json file

* Remove e2e directory

IraCherginyshka committed:
Change prefix app => atg (#3)

IraCherginyshka committed:
Feature/ATG-5 Add EsLint (#4)

* Add EsLint to project

* proj => AwesomeTetrisGame; remove e2e settings

IraCherginyshka committed:
Create scss file with variables (#5)

* Create variables file

* Resolve

IraCherginyshka committed:
Feature/ATG-10 Create main field with general styles (#6)

* Create components (Header, Board, Option, Controls, Information, Leaderboard). Add icons to images folder.

* Create components (Header, Board, Option, Controls, Information, Leaderboard). Add icons to images folder.

* Add variables to styles

* Resolve conversation. Comments.

* Fix EsLint

* Delete .prettierrc.js

Delete unnecessary file

Commit:
Commit

Commit:
Commit

Commit:
Commit

IraCherginyshka committed:
Feature/ATG-5 (#7)

* Add EsLint to project

* proj => AwesomeTetrisGame; remove e2e settings

Commit:
Merge branch 'dev' of https://github.com/IraCherginyshka/AwesomeTetrisGame into dev

IraCherginyshka committed:
Bugfix/ATG-48 Prettier file (#11)

* Commit

* Commit

* Move file

IraCherginyshka committed:
Feature/ATG-12 Controls component (#10)

* Commit

* Commit

* Create GameControls component

* Delete .prettierrc.js

Commit:
Resolve conflicts

IraCherginyshka committed:
Feature/ATG-9 Header component (#8)

* Commit

* Commit

* Create Header html-template

* Delete .prettierrc.js

* Run eslint

IraCherginyshka committed:
Feature/ATG-14 Information field (#9)

* Commit

* Commit

* Create information field

* Delete

* Run Eslint

Commit:
Merge branch 'dev' of https://github.com/IraCherginyshka/AwesomeTetrisGame into dev

IraCherginyshka committed:
Bugfix/ATG-49 hooks: 'pre-push' => 'pre-commit' (#12)

* Commit

* Commit

* pre-push => pre-commit; Delete --fix from script eslint

* 'Resolve'

Commit:
Merge branch 'dev' of https://github.com/IraCherginyshka/AwesomeTetrisGame into dev

Commit:
Sync wit dev

IraCherginyshka committed:
Update README file with information about way of work (#16)

IraCherginyshka committed:
Change styles controls, header (#15)

IraCherginyshka committed:
Feature/ATG-47 Routing  (#13)

* Add routing. Create page-not-found component

* Delete page-not-found component, fix links

IraCherginyshka committed:
Feature/ATG-13 State-controls Component (#14)

* Create html template with styles state-controls-component

* Fix state controls (order), add cursor: pointer to buttons

IraCherginyshka committed:
Create constants for all board-component (#20)

Commit:
Create class Block

Commit:
Rename file

IraCherginyshka committed:
Feature/ATG-60 Enums (figures colors) (#18)

* Create colors enum for figures

* Fix export. Rename files

IraCherginyshka committed:
Feature/ATG-58 Figures Matrix (#19)

* Create matrix for all figures

* Rename file
Delete default export

Commit:
Rename file
type function

Commit:
Rename file

Commit:
Merge branch 'dev' of https://github.com/IraCherginyshka/AwesomeTetrisGame into feature/ATG-59

Commit:
Merge branch 'feature/ATG-59' into dev

IraCherginyshka committed:
Feature/ATG-63 Create canvas, class Figure, styles to board, logic game (move down) (#22)

* Fix export. Rename files

* Rename file

* Rename file
         Add types

* Add types

* Resolve code review

* Resolve code review

maksymfedenko committed:
Improve eslint config

IraCherginyshka committed:
Merge pull request #24 from IraCherginyshka/improvement/eslint-config

Improve eslint config

Commit:
New rules (Eslint)

IraCherginyshka committed:
Feature/ATG-62 Random figures (#23)

* Random figures with color enum

* Resolve comment
New private function in Figure model

IraCherginyshka committed:
Feature/ATG-68 Service for game state  (#25)

* Create game-state service for whole project
Management information about state of game
Pause/Reset buttons
Change folders for controls

* unsubscribe

* Fixed code

* Private method

* fix return drawBoard()

IraCherginyshka committed:
Feature/ATG-69 Keyboard event (left, right) (#26)

* Create GameMovementService
Add HostListener to controls component
Create enum with figures-movement

* onDestroy

* Resolve
add .pipe
rename function

IraCherginyshka committed:
new favicon (#30)

IraCherginyshka committed:
Feature/ATG-44 Figure rotation (#29)

* Figure rotation

* add preventDefault;

* remove method to Figure

IraCherginyshka committed:
new styles (#31)

IraCherginyshka committed:
ATG-21 Pause/play (#32)

* Implementation of a pause/play in the game

* resolve conversation

IraCherginyshka committed:
Feature/ATG-33 Next figure (#33)

* create board model
update gameService
logic nextFigure component

* resolve conversations

add abstract class Block

* abstract method

* fix service

IraCherginyshka committed:
Feature/ATG-64 Update board. Checking collision detection.  (#34)

* Update board,
check collision detection

* changed speed

* add types

* resolve conversation

* resolve conversation

IraCherginyshka committed:
Press key DOWN => increase speed (#35)

IraCherginyshka committed:
fix showFigure function in model (add if statement) (#37)

IraCherginyshka committed:
Feature/ATG-37  Delete filled line (#36)

* Delete filled line

* resolve conversation (improve function findFilledLine)

IraCherginyshka committed:
Feature/ATG-66 Display LOST game (#38)

* Create lostGameSubject
isLostGame condition

* Delete 'value'

* update game-board template

* resolve conversation

IraCherginyshka committed:
update README file (#39)

IraCherginyshka committed:
key press P, R, Enter (#42)

IraCherginyshka committed:
Buttons l/r/rot/down (#40)

improved board functionality

IraCherginyshka committed:
Bugfix (feature/ATG-75) Fix function findFilledLine in model (#41)

* Fix function findFilledLine in model
add condition

* resolve conversation

IraCherginyshka committed:
Bugfix/ATG-76 Fix key press Play (#43)

* fix key press play

* resolve conversation

IraCherginyshka committed:
Feature/ATG-36 Increase score (#44)

* Increase score
create new file with constants for game information

* rename function

IraCherginyshka committed:
Feature/ATG-34 Increase Level  (#46)

* Update information about current level => increase speed

Improve Reset game

Use async pipes in information component

* Resolve conversation
private method => public
use BehaviorSubject
create Interface for gameStats

IraCherginyshka committed:
Feature/ATG-85 LogIn/SignUp forms (#47)

* Create Login form
Create Registration form

* Resolve conversation

IraCherginyshka committed:
Feature/ATG-85 Improved form//validation  (#48)

* Create Login form
Create Registration form

* Resolve conversation

* Improved verification for confirm password

* eslint

* fix

IraCherginyshka committed:
Set node, mongo, express (#49)

Add functionality sign-up/login/log-out

IraCherginyshka committed:
Feature ATG/92 Toaster (#50)

* add toaster
add new styles

* delete .then()

IraCherginyshka committed:
Feature/ATG-93 Common leaderboard layout (#51)

* create a common leaderboard layout

* Resolve conversation (semantics)

IraCherginyshka committed:
Feature/ATG-95 Leaderboard //best, current (#52)

* Leaderboard functionality
New route /logout

* resolve conversation

Commit:
port

Commit:
test

Commit:
build dist

Commit:
add dist

Commit:
build

Commit:
build
static
eslint
endpoint

IraCherginyshka committed:
remove routes (#53)

oleksandrstarov commented on server/app.js:
```diff
@@ -0,0 +1,105 @@
+};
+
+app.post('/sign_up', (req, res) => {
+  const user = new User(req.body);
```
no variable is required

oleksandrstarov commented on server/app.js:
```diff
@@ -0,0 +1,105 @@
+    })
+    .catch(() => {
+      return errorHandler(res);
+    });
```
should be moved to db.js, db.createUser(req.body).then().catch()

oleksandrstarov commented on src/app/app.module.ts:
```diff
@@ -0,0 +1,63 @@
+import { HTTP_INTERCEPTORS, HttpClientModule } from '@angular/common/http';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+
+import { ToastContainerModule, ToastrModule } from 'ngx-toastr';
```
move one line up

oleksandrstarov commented on src/app/components/game-board/game-board.component.scss:
```diff
@@ -0,0 +1,18 @@
+  position: absolute;
+  width: $board-width + 2px;
+  height: $board-height + 3px;
+  background-color: rgba($primary-color, 0.85);
```
add variable to .85

oleksandrstarov commented on src/app/components/game-board/game-board.component.ts:
```diff
@@ -0,0 +1,253 @@
+  }
+
+  private rotateFigure(figureMatrix: FiguresColors[][]): FiguresColors[][] {
+    const reverseMatrix = [...figureMatrix];
```
[...figureMatrix].reverse();

oleksandrstarov commented on src/app/components/game-board/game-board.component.ts:
```diff
@@ -0,0 +1,253 @@
+    const newFigure = new FigureModel();
+    const newBoard = new BoardModel(this.ctx, false);
+
+    this.currentMatrix = newFigure.showFigure(
```
newFigure >> new FigureModel()

oleksandrstarov commented on src/app/components/game-board/game-board.component.ts:
```diff
@@ -0,0 +1,253 @@
+
+  private playGame(): void {
+    this.timeInterval = window.setInterval(() => {
+      if (this.checkCollisionDetection(0, this.currentFigure)) {
```
add const hasCollision = this.checkCollisionDetection(0, this.currentFigure);

oleksandrstarov commented on src/app/components/game-board/game-board.component.ts:
```diff
@@ -0,0 +1,253 @@
+        this.lineWithFigure += 1;
+      } else if (
+        !this.checkCollisionDetection(0, this.currentFigure) &&
+        this.lineWithFigure === 0
```
!lineWith...

oleksandrstarov commented on src/app/components/game-board/game-board.component.ts:
```diff
@@ -0,0 +1,253 @@
+    );
+    this.gameService.updateFigures();
+    this.redrawBoard();
+    clearInterval(this.timeInterval);
```
stopGame()

oleksandrstarov commented on src/app/components/game-board/game-board.component.ts:
```diff
@@ -0,0 +1,253 @@
+    });
+    this.gameService.updateFigures();
+    this.setInitialBoardState();
+    clearInterval(this.timeInterval);
```
stopGame()

oleksandrstarov commented on src/app/components/game-controls/game-controls.component.ts:
```diff
@@ -0,0 +1,75 @@
+export class GameControlsComponent implements OnInit, OnDestroy {
+  public moveStep = FiguresMovement;
+
+  private isPlaying: boolean;
```
separate public/private

oleksandrstarov commented on src/app/components/game-header/game-header.component.html:
```diff
@@ -0,0 +1,16 @@
+  <atg-player-profile *ngIf="userIsAuthenticated; else login"></atg-player-profile>
+  <div class="wrap-links">
+    <nav class="bar-links container">
+      <a class="header-link" [routerLink]="['/login']">login</a>
```
please check routerLinkActive

IraCherginyshka replied:
Thanks

IraCherginyshka committed:
Bugfix/ATG-97 Fix//  game starts not from first level after Game over (#54)

* fixed Pause
game started from level 1 after game over

* add dist folder

IraCherginyshka committed:
Feature/atg 99 Validation forms (#56)

* add validation to age (min 6)
add padding
install moment
add validation to country (min/max length)

* add new files

* commit

* lines 10 => 5

* resolve conversation
text notification, naming

* add dist folder

IraCherginyshka committed:
Feature/ATG-113 New styles (#57)

* new styles

* add dist

Commit:
delete dist folder

IraCherginyshka committed:
update readme file (#58)

oleksandrstarov commented on src/app/components/game-header/game-header.component.ts:
```diff
@@ -0,0 +1,40 @@
+          this.toastrService.success('Welcome to AwesomeTetrisGame!');
+          this.userIsAuthenticated = true;
+        } else {
+          this.userIsAuthenticated = false;
```
can be this.userIsAuthenticated = !!user

Commit:
resolve conversation:
no variable is required (user, result) in app.js

Commit:
resolve conversation:
Change import: order, empty lines

Commit:
resolve conversation:
Update styles with variables

Commit:
resolve conversation:
[...figureMatrix].reverse();

Commit:
resolve conversation:
make board component more semantic

Commit:
resolve conversation:
separate public/private

Commit:
fix noCollision

Commit:
resolve conversation
can be this.userIsAuthenticated = !!user

Commit:
resolve conversation
add routerLinkActive

Commit:
Create LocalStorage ENUM

Commit:
Rename app.js => server.js

Commit:
rename interfaces
remove functionality to db
delete information about password from localStorage

IraCherginyshka committed:
Merge pull request #61 from IraCherginyshka/feature/ATG-118

Feature/ATG-118 From code review
